modules/programs/yt-dlp.nix

```diff
@@ -6,6 +6,11 @@ let
 
   cfg = config.programs.yt-dlp;
 
+  renderedSettings = settings:
+    concatStringsSep "\n" (mapAttrsToList
+      (name: value: if isBool value then "--${name}" else "--${name} ${value}")
```

Comment:
If you set

    embed-thumbnail = false;

then I guess the generated file should not contain `--embed-thumbnail` at all?